src/app/widgets/JogControl/Keypad.jsx

```diff
@@ -155,6 +159,10 @@ class Keypad extends PureComponent {
         const canClickXY = canClickX && canClickY;
         const canClickZ = canClick && _includes(axes, 'z');
 
+        const { ROTARY } = WORKSPACE_MODE;
+        const rotary = store.get('workspace.mode') === ROTARY;
+        const firmwareType = get(reduxStore.getState(), 'controller.type');
```

Comment:
Don't see the point of using `reduxStore.getState()` here, better practice would be to retrieve the value and send it as a prop in the connect HOC, for reference you can look at the component export in `src/app/widgets/Spindle/index.jsx`